### DISCUSSION TITLE
0.3.0-stable has been released!

### DISCUSSION THREAD

**Discussion by mihe** [godot-jolt/godot-jolt#467] [Announcements]
This release supports [**Godot 4.0.3**](https://github.com/godotengine/godot/releases/tag/4.0.3-stable).

## Known major issues

- Extension classes cause exception when used from C# ([#270](https://github.com/godot-jolt/godot-jolt/issues/270))
- The `move_and_collide` parameter `recovery_as_collision` is always `true` ([#337](https://github.com/godot-jolt/godot-jolt/issues/337))

## Changelog

These are the notable changes that have been made since [0.2.3-stable](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.2.3-stable) was released. You can also find a list of all the commits [here](https://github.com/godot-jolt/godot-jolt/compare/v0.2.3-stable...v0.3.0-stable).

Any breaking changes are denoted with ⚠️.

### Changed

- ⚠️ Changed collision layers and masks to behave as they do in Godot Physics, allowing for asymmetrical collisions, where the body whose mask does not contain the layer of the other body effectively gets infinite mass and inertia in the context of that collision.

### Added

- Added new project setting, "Use Shape Margins", which when disabled leads to all shape margins being ignored and instead set to 0, at a slight performance cost.
- Added new project setting, "Areas Detect Static Bodies", to allow `Area3D` to detect overlaps with static bodies (including `RigidBody3D` using `FREEZE_MODE_STATIC`) at a potentially [heavy performance/memory cost](https://github.com/godot-jolt/godot-jolt/blob/master/docs/settings.md#jolt-3d).

### Fixed

- Fixed issue where a `RigidBody3D` using `FREEZE_MODE_KINEMATIC` wouldn't have its `_integrate_forces` method called when monitoring contacts.
- Fixed issue where scaling bodies/shapes with negative values would break them in various ways.
- Fixed issue where `CharacterBody3D` platform velocities would always be zero.
- Fixed issue where the velocity of kinematic colliders would always be zero in `_physics_process`.

<hr /><em>This discussion was created from the release <a href='https://github.com/godot-jolt/godot-jolt/releases/tag/v0.3.0-stable'>0.3.0-stable</a>.</em>

**Comment by mgiraud** [comment]
Can you tell us your plans about godot 4.1.0 compatibility ?

**Reply by mihe** [reply]
I've had to upgrade to Godot 4.1 a couple of times already, as part of some Godot pull requests that I had to make, and it's thankfully a fairly easy transition for Godot Jolt.

Once Godot 4.1.0-stable is published (presumably some time next week) then a new version of Godot Jolt (likely 0.4.0) will follow very shortly after.

There have been some major improvements to GDExtension as part of Godot 4.1, which will make GDExtension much more forward-compatible, meaning Godot Jolt should hopefully just work with any future version of Godot 4, without me needing to make a new version right away for every release of Godot. 🤞

There is one breaking change in Godot 4.1, in terms of physics, which is that contact positions have changed from being relative global coordinates to instead being absolute global coordinates, to line up with how the 2D physics behaves, so I will be making the same breaking change in Godot Jolt. You can read more about it [here](https://github.com/godotengine/godot/pull/76088).

**Reply by Gamepro5** [reply]
4.1 just came out 🎉. It completely broke jolt. Hopefully they do as you say and make it more forward compatible.

**Reply by mihe** [reply]
Neat. You should see 0.4.0-rc1 here on GitHub in a few hours, which will be compatible with Godot 4.1.

**Comment by Gamepro5** [comment]
> Added new project setting, "Use Shape Margins", which when disabled leads to all shape margins being ignored and instead set to 0, at a slight performance cost.

Thank you!! ;)
